### ISSUE TITLE
ladda-label and ladda-spinner disappear after switching locales

### ISSUE THREAD

**Issue by g-p-g** [jsdir/react-ladda#16]
See the full example below to replicate the problem.

Basically, when using react-intl and react-router it's common to call `React.render(...)` at a later moment to switch locales. By doing this as in the code below, the elements ladda-label and ladda-spinner go missing causing the animations to stop working. Although this involves several packages, so far it seems the problem is in react-ladda.

```
var React = require('react');
var Router = require('react-router');
var ReactIntl = require('react-intl');
var Route = Router.Route;
var LaddaButton = require('react-ladda');


var locale = {
  en: {
    locales: ["en"],
    messages: {"button": "hello"}
  },
  es: {
    locales: ["es"],
    messages: {"button": "hola"}
  },
  active: 'en'
};

var App = React.createClass({
  mixins: [ReactIntl.IntlMixin],

  _reRender: function() {
    locale.active = 'es';
    App.appRender();
  },

  render: function() {
    return (
      <div>
        <a href="#" onClick={this._reRender}>Render</a>
        <Router.RouteHandler/>
      </div>
    );
  }
});

var Home = React.createClass({
  mixins: [ReactIntl.IntlMixin],
  render: function() { return <Button />; }
});

var Button = React.createClass({
  mixins: [ReactIntl.IntlMixin],

  getInitialState: function() {
    return {submitted: false};
  },

  _submit: function() {
    this.setState({submitted: true});
    setTimeout(function() {
      this.setState({submitted: false});
    }.bind(this), 1000);
  },

  render: function() {
    return (
      <form onSubmit={this._submit}>
        <LaddaButton style="zoom-in" active={this.state.submitted}>
          <button type="submit">{this.getIntlMessage('button')}</button>
        </LaddaButton>
      </form>
    );
  }
});


var routes = (
  <Route handler={App}>
    <Route name="home" path="/" handler={Home}/>
  </Route>
);

App.CurrentHandler = null;
App.appRender = function() {
  console.log('appRender', App.CurrentHandler);
  if (App.CurrentHandler) {
    React.render(<App.CurrentHandler {...locale[locale.active]} />,
                 document.body);
  }
};

Router.run(routes, function(Handler) {
  App.CurrentHandler = Handler;
  App.appRender();
});
```

**Comment by g-p-g** [comment]
A workaround is to manually create a ladda-label as in `<button type="submit"><span className="ladda-label">{this.getIntlMessage('button')}</span></button>`

**Comment by jsdir** [comment]
I just wrote [a test](https://github.com/jsdir/react-ladda/blob/cec597e44df47180ed869635ef3671c69faa3d9d/__tests__/LaddaButton-test.js#L62) to check the persistence of the button elements given the multiple `React.render` invocations, and it seems to be working fine. What version of React are you using?

**Comment by g-p-g** [comment]
Just so I know, did you run the exact sample above and it works for you?

**Comment by jsdir** [comment]
Sorry for the delay, I ran the above example and I got exactly what you were describing. I looked into it further and found that React was updating the DOM after Ladda (had already changed it)[https://github.com/hakimel/Ladda/blob/e10abaf7c5dd96adce0719019ca6756f5e4a1995/js/ladda.js#L48]. If we are going to support changing the button contents through React after the component is initialized with Ladda, the new content may have to be explicitly rendered into the relocated DOM element with `React.render`.

**Comment by g-p-g** [comment]
Thanks for giving it a try. Would it be an option to stop accepting arbitrary children for LaddaButton? That seems to simplify things a lot and eliminate this kind of problem (it takes, let's say, a message prop which  gets rendered correctly), other props would be passed down to a <button .. /> that LaddaButton creates instead of the user.

I have no idea if people are using LaddaButton in different ways where you do way more than just a button inside it, which would make the idea above unworkable.

**Comment by jsdir** [comment]
This would be a good solution since we are never entirely sure that Ladda will continue to relocate DOM elements the way it currently does. Giving `LaddaButton` complete ownership of the `button` is also more developer-friendly, especially since the only practical customization of the button that I can see would be exclusively through a `message` or `title` prop. I'll work on releasing this as a breaking API change.

**Comment by ultrafez** [comment]
Just to add my two pennies worth, I've been using react-ladda by including a Bootstrap button as a child (just like a regular button, but with the `btn` and `btn-primary` classes added). I'm encountering the same issue where the progress div and label span disappear, but what I'm using as a workaround for the moment is this:

    <LaddaButton>
        <button type="button" className="btn btn-primary">
            <span className="ladda-label">{buttonLabel}</span>
            <span className="ladda-spinner"></span>
        </button>
    </LaddaButton>

This survives React's re-rendering and so the react-ladda library works properly.

If you're planning to make a new version that gives `LaddaButton` complete control of its rendering, can I suggest that you use `{...props}` on the `<button>` element that you render, to still allow adding custom classes?

**Comment by jsdir** [comment]
@ultrafez: I've implemented this fix in `react-ladda@3.0.0`

**Comment by ultrafez** [comment]
That works great now. Thanks!
